concordium-consensus/src/Concordium/KonsensusV1/TreeState/StartUp.hs

```diff
@@ -209,3 +231,88 @@ loadSkovData _runtimeParameters = do
+      TimeMonad m
+    ) =>
+    m ()
+loadCertifiedBlocks = do
+    certBlocks <- LowLevel.lookupCertifiedBlocks
+    mapM_ loadCertBlock certBlocks
+    oLastTimeout <- use $ persistentRoundStatus . prsLatestTimeout
+    forM_ oLastTimeout $ \lastTimeout -> do
+        curRound <- use $ roundStatus . rsCurrentRound
+        when (tcRound lastTimeout >= curRound) $ do
+            highCB <- use $ roundStatus . rsHighestCertifiedBlock
+            unless
+                ( cbRound highCB < tcRound lastTimeout
+                    && cbRound highCB >= tcMaxRound lastTimeout
+                    && cbEpoch highCB >= tcMaxEpoch lastTimeout
+                    && cbEpoch highCB <= 2 + tcMinEpoch lastTimeout
+                )
+                $ throwM . TreeStateInvariantViolation
+                $ "Missing certified block consistent with last timeout certificate"
+            roundStatus . rsPreviousRoundTimeout
+                .= Present
+                    RoundTimeout
+                        { rtTimeoutCertificate = lastTimeout,
+                          rtCertifiedBlock = highCB
+                        }
+            roundStatus . rsCurrentRound .= tcRound lastTimeout + 1
+    rs <- use roundStatus
+    let expectedCurrentRound
+            | Present prevTO <- rs ^. rsPreviousRoundTimeout = 1 + tcRound (rtTimeoutCertificate prevTO)
+            | otherwise = 1 + cbRound (rs ^. rsHighestCertifiedBlock)
+    unless (expectedCurrentRound == rs ^. rsCurrentRound) $
+        throwM . TreeStateInvariantViolation $
+            "The current round does not match the expected round."
+  where
+    loadCertBlock (storedBlock, qc) = do
+        blockPointer <- mkBlockPointer storedBlock
+        cacheBlockState (bpState blockPointer)
+        blockTable . liveMap . at' (getHash blockPointer) ?=! MemBlockAlive blockPointer
+        addToBranches blockPointer
+        forM_ (blockTransactions blockPointer) $ \tr -> do
+            -- Add transactions to the transaction table as 'TVer.TrustedSuccess', since they
+            -- occur in blocks that have already been checked.
+            let verRes = TVer.Ok TVer.TrustedSuccess
+            added <-
+                transactionTable
+                    %%=! TT.addTransaction tr (TT.commitPoint (blockRound blockPointer)) verRes
+            unless added $
+                throwM . TreeStateInvariantViolation $
+                    "Transaction in certified block cannot be added to transaction table"
+            addPendingTransaction tr
+        roundStatus . rsHighestCertifiedBlock
+            .= CertifiedBlock
+                { cbQuorumCertificate = qc,
+                  cbQuorumBlock = blockPointer
+                }
+
+        curRound <- use $ roundStatus . rsCurrentRound
+        when (blockRound blockPointer >= curRound) $ do
+            roundStatus
+                %=! (rsPreviousRoundTimeout .~ Absent)
+                . (rsCurrentRound .~ blockRound blockPointer + 1)
```

Comment:
Shouldn't this be `qcRound qc + 1`  ?

Reply:
Since the QC is for the block, these should be the same.